test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/net/URLDecoderTest.scala

```diff
@@ -0,0 +1,74 @@
+package org.scalajs.testsuite.javalib.math
+
+import java.net.URLDecoder
+import java.io.UnsupportedEncodingException
+import org.scalajs.jasminetest.JasmineTest
+
+object URLDecoderTest extends JasmineTest {
+
+  describe("URLDecoderTest"){
+
+    it("returns the empty string unchanged"){
+      expect(URLDecoder.decode("", "utf-8")).toEqual("")
+    }
+
+    it("transforms + in spaces"){
+      expect(URLDecoder.decode("a+b+c", "utf-8")).toEqual("a b c")
+    }
+
+    it("transforms encoded caracters correctly"){
+      expect(URLDecoder.decode("a%20b%20c", "utf-8")).toEqual("a b c")
+    }
+
+    it("throws IllegalArgumentException for invalid hex characters"){
+      expect {
+        try {
+          URLDecoder.decode("a%b%c", "utf-8")
+          false
+        } catch {
+          case _: IllegalArgumentException => true
+          case _: Throwable => false
+        }
+      }.toBeTruthy
+    }
+
+    it("throws UnsupportedEncodingException for invalid encodings"){
+      expect {
+        try {
+          URLDecoder.decode("a+b+c", "sebastian")
+          false
+        } catch {
+          case _: UnsupportedEncodingException => true
+          case _: Throwable => false
+        }
+      }.toBeTruthy
+    }
+
+    it("works with iso-8859-1"){
+      expect(URLDecoder.decode("a%20b%20c", "iso-8859-1")).toEqual("a b c")
```

Comment:
Well you should test some ISO-8859-1-specific code units, then, i.e., characters in the range 128-255. That line would "fail to fail" if `decode()` ignored the second argument and used UTF-8 regardless.